### PR TITLE
Graphql Auth Migration Done

### PR DESCRIPTION
**Pull request by KristianWEB**
* Implemented Issue #97 
* Implemented Issue #98 
* Implemented Issue #99 

I've also had to integrate GraphQL first before implementing them ( using `apollo-client` but not the full package `apollo-boost`. I'm using the separate packages that build it up so I have more access to the setup )

### CODE REVIEW

File: src/App/actions/auth.js [hunk 1]
```diff
@@ -12,86 +10,52 @@ import {
...
-  const config = {
-    headers: {
-      "Content-Type": "application/json"
-    }
-  };
-
-  const body = JSON.stringify({ username, email, password });
 
+// Register User
+export const register = registerData => async dispatch => {
```
Comment (vinodh99) [inline src/App/actions/auth.js]: why do we still have the async without await?

File: src/App/actions/auth.js [hunk 2]
```diff
@@ -12,86 +10,52 @@ import {
...
 // Login User
-export const login = (email, password, remember) => async dispatch => {
-  const config = {
-    headers: {
-      "Content-Type": "application/json"
-    }
-  };
-
-  const body = JSON.stringify({ email, password });
+export const login = (loginData, remember) => async dispatch => {
```
Comment (vinodh99) [inline src/App/actions/auth.js]: Same here

File: src/App/actions/auth.js [hunk 3]
```diff
@@ -12,86 +10,52 @@ import {
...
 import setAuthToken from "../utils/setAuthToken";
 
-const BaseURL = "https://osd-sidekick.herokuapp.com/api/auth";
-
-// const LocalURL = "http://localhost:8080/api/auth";
-
-// Load User
-
-export const loadUser = () => async dispatch => {
+export const loadUser = currentUser => async dispatch => {
```
Comment (vinodh99) [inline src/App/actions/auth.js]: same here

File: src/App/actions/auth.js (hunk 1 above)
Comment (KristianWEB) [reply src/App/actions/auth.js]: I forgot to remove `async` keyword. I'll change that in a second, thanks!

**Pull request merged**